driver_android.go

```diff
@@ -64,6 +87,31 @@ static char* initAudioTrack(uintptr_t java_vm, uintptr_t jni_env,
       (*env)->GetStaticIntField(
           env, android_media_AudioFormat,
           (*env)->GetStaticFieldID(env, android_media_AudioFormat, "ENCODING_PCM_16BIT", "I"));
+  jint android_media_AudioAttributes_USAGE_UNKNOWN = 0;
+  jint android_media_AudioAttributes_CONTENT_TYPE_UNKNOWN = 0;
+  jint android_media_AudioAttributes_FLAG_LOW_LATENCY = 0;
+  if (hasAPI24) {
+    android_media_AudioAttributes_USAGE_UNKNOWN =
+        (*env)->GetStaticIntField(
+            env, android_media_AudioAttributes,
+            (*env)->GetStaticFieldID(env, android_media_AudioAttributes, "USAGE_UNKNOWN", "I"));
+    android_media_AudioAttributes_CONTENT_TYPE_UNKNOWN =
+        (*env)->GetStaticIntField(
+            env, android_media_AudioAttributes,
+            (*env)->GetStaticFieldID(env, android_media_AudioAttributes, "CONTENT_TYPE_UNKNOWN", "I"));
+    // FLAG_LOW_LATENCY only exists since API 24, 7.0 Nougat
+    const jfieldID fieldFlagLowLatency = (*env)->GetStaticFieldID(env, android_media_AudioAttributes, "FLAG_LOW_LATENCY", "I");
+    if ((*env)->ExceptionCheck(env)) {
```

Comment:
When does this exception appear?

Reply:
This exception occurs on GetStaticFieldID when the field does not exist. According to [the documentation here](https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html), the function GetStaticFieldID returns NULL if the field does not exist, but it also says that it throws NoSuchFieldError. Maybe better to check for both?

Reply:
Ah OK, the first resetting `hasAPI24` is done with checking the API ver is 21, not 24.

Reply:
> Maybe better to check for both?

No, i think checking the existence of the exception should be fine.

Reply:
Ah I see. Yes, that if statement is not clear at all. Maybe it would be better to use a second boolean `hasAPI21`?

Reply:
Yeah, adding `hasAPI21` would make the code less confusing.